### PR TITLE
Add X-Frame-Options and CSP rule

### LINKED ISSUE #1466: Resolve X-Frame-Options Header Alert from Security Scan
Results from OWASP ZAP scan identified potential for "Clickjacking" because of lack of X-Frame-Options header iin the HTTP response. For full report, [download the HTML file from this Slack thread](https://hpe.slack.com/archives/GPPLUK6LR/p1614878349091400):
Screen Shot 2021-03-04 at 11 19 15 AM

Related to: #1290

### PR DESCRIPTION
**Pull request by jcfilben**
#### What does this PR do?
From issue #1466:
> Results from OWASP ZAP scan identified potential for "Clickjacking" because of lack of X-Frame-Options header in the HTTP response.

I added the following rule to the next.config.js file:
```javascript
source: '/(.*)?',
        headers: [
          {
            key: 'X-Frame-Options',
            value: 'SAMEORIGIN',
          }
        ]
```
This allows the page to only be displayed in a frame on the same origin as the page itself to prevent clickjacking attacks.

I also added the following CSP rule which tells the browser what domains are allowed:
```javascript
<meta httpEquiv="Content-Security-Policy" content="default-src 'self' 'unsafe-eval'; 
        style-src 'self' *.hpe.com/hfws-static/slim/css/ 'unsafe-inline';
        connect-src 'self' *.githubusercontent.com/grommet/hpe-design-system/ https://www.google-analytics.com;
        media-src 'self' https://d3hq6blov2iije.cloudfront.net/media/HPE+Design+System-v3.mp4;
        img-src 'self' https://www.google-analytics.com;
        script-src-elem 'self' *.hpe.com https://www.google-analytics.com/analytics.js;
        font-src *.hpe.com hpefonts.s3.amazonaws.com *.cloudfront.net/fonts/;
        object-src 'none';"></meta>
```
Notes:
**‘unsafe-eval’**
That is used in the `<meta>` above, allows the use of eval() and similar methods for creating code from strings. It needed to be added in default-src because webpack uses eval().
eval() can potentially be unsafe because it executes the code it is passed with the privileges of the caller. ‘unsafe-eval’ is only allowed on 'self' (only same origin, no external sources).
Without ‘unsafe-eval’ the following errors occur:
image

**‘unsafe-inline’**
allows the use of inline resources, such as inline <script> elements, javascript: URLs, inline event handlers, and inline <style> elements.
This can be unsafe because it creates the potential for code to be injected into inline tags. ‘unsafe-inline’ is only allowed on the following: 'self' and *.hpe.com/hfws-static/slim/css/
Example error when ‘unsafe-inline’ is not included:
image

More info about clickjacking, CSP, and X-Frame-Options: https://owasp.org/www-community/attacks/Clickjacking

#### What testing has been done on this PR?
Tested CSP with https://csp-evaluator.withgoogle.com/
Tested in browser and tested with testcafe

#### How should this be manually tested?
Using any of the methods above, and checking that no additional resources are required that aren't included in the CSP rule

#### What are the relevant issues?
Closes #1466

#### Screenshots (if appropriate)
Screen Shot 2021-03-12 at 12 21 08 PM

#### Should this PR be mentioned in Design System updates?
No

#### Is this change backwards compatible or is it a breaking change?
Backwards compatible

### CODE REVIEW

**Review by ShimiSun — APPROVED** [review]
LGTM!! Just need to fix the tests and this PR is good to go.

**Comment by taysea** [comment]
All looks good here except the e2e test log seems to be saying that it can't access the Applitools API key. I'll look into that this morning!

File: aries-site/src/components/seo/Meta.js [hunk 1]
```diff
@@ -89,6 +89,16 @@ export const Meta = ({ title, description, canonicalUrl, socialImageUrl }) => {
       <meta key="segment" name="segment" content={segment} />
       <meta key="lifecycle" name="lifecycle" content={lifecycle} />
       <meta key="page_content" name="page_content" content={pageContent} />
+      <meta
+httpEquiv="Content-Security-Policy" 
+        content="default-src 'self' 'unsafe-eval'; 
+        style-src 'self' *.hpe.com/hfws-static/slim/css/ 'unsafe-inline';
+        connect-src 'self' *.githubusercontent.com/grommet/hpe-design-system/ https://www.google-analytics.com;
```
Comment (taysea) [inline aries-site/src/components/seo/Meta.js:96]:
```suggestion
        connect-src 'self' *.githubusercontent.com/grommet/hpe-design-system/ https://www.google-analytics.com https://www.github.com;
```

File: aries-site/src/components/seo/Meta.js [hunk 2]
```diff
@@ -89,6 +89,16 @@ export const Meta = ({ title, description, canonicalUrl, socialImageUrl }) => {
       <meta key="segment" name="segment" content={segment} />
       <meta key="lifecycle" name="lifecycle" content={lifecycle} />
       <meta key="page_content" name="page_content" content={pageContent} />
+      <meta
+httpEquiv="Content-Security-Policy" 
+        content="default-src 'self' 'unsafe-eval'; 
+        style-src 'self' *.hpe.com/hfws-static/slim/css/ 'unsafe-inline';
+        connect-src 'self' *.githubusercontent.com/grommet/hpe-design-system/ https://www.google-analytics.com https://www.github.com;
```
Comment (taysea) [inline aries-site/src/components/seo/Meta.js:96]:
```suggestion
        connect-src 'self' *.githubusercontent.com/grommet/hpe-design-system/ https://www.google-analytics.com https://www.github.com https://www.applitools.com;
```

**Comment by jcfilben** [comment]
Closing in favor of #1493

**Pull request closed without merging**